libuavcan/include/uavcan/transport/can_io.hpp

```diff
@@ -29,94 +30,81 @@ struct UAVCAN_EXPORT CanRxFrame : public CanFrame
 
     CanRxFrame()
         : iface_index(0)
-    { }
+    {
+    }
 
 #if UAVCAN_TOSTRING
     std::string toString(StringRepresentation mode = StrTight) const;
 #endif
 };
 
-
-class UAVCAN_EXPORT CanTxQueue : Noncopyable
+struct CanTxQueueEntry // Not required to be packed - fits the block in any case
 {
-public:
-    enum Qos { Volatile, Persistent };
-
-    struct Entry : public LinkedListNode<Entry>  // Not required to be packed - fits the block in any case
+    MonotonicTime deadline;
+    const CanFrame frame;
+    CanIOFlags flags;
+
+    CanTxQueueEntry(const CanFrame& arg_frame, const MonotonicTime arg_deadline, CanIOFlags arg_flags)
+        : deadline(arg_deadline)
+        , frame(arg_frame)
+        , flags(arg_flags)
```

Comment:
The codebase is a bit inconsistent within itself, it is a known problem. Member initializers shall be formatted as follows:

```c++
CanTxQueueEntry(const CanFrame& arg_frame, const MonotonicTime arg_deadline, CanIOFlags arg_flags) :
    deadline(arg_deadline),
    frame(arg_frame),
    flags(arg_flags)
```